lfs/diff_index_scanner.go

```diff
@@ -0,0 +1,191 @@
+package lfs
+
+import (
+	"bufio"
+	"strconv"
+	"strings"
+
+	"github.com/git-lfs/git-lfs/errors"
+)
+
+// Status represents the status of a file that appears in the output of `git
+// diff-index`.
+//
+// More information about each of its valid instances can be found:
+// https://git-scm.com/docs/git-diff-index
+type Status rune
```

Comment:
I'm cool with `lfs.DiffIndexStatus` for now. My rough plan was to move all gitscanner stuff to a separate package once all the refactoring is done. I don't think a `git/diffindex` package is necessary right now.